--- a/src/drv_imap.c	2021-12-03 10:56:16.000000000 +0000
+++ b/src/drv_imap.c	2022-03-04 02:37:25.000000000 +0000
@@ -2195,6 +2195,7 @@
 	{ SASL_CB_USER,     NULL, NULL },
 	{ SASL_CB_AUTHNAME, NULL, NULL },
 	{ SASL_CB_PASS,     NULL, NULL },
+        { SASL_CB_OAUTH2_BEARER_TOKEN, NULL, NULL },
 	{ SASL_CB_LIST_END, NULL, NULL }
 };
 
@@ -2212,6 +2213,7 @@
 			val = ensure_user( srvc );
 			break;
 		case SASL_CB_PASS:
+                case SASL_CB_OAUTH2_BEARER_TOKEN:
 			val = ensure_password( srvc );
 			break;
 		default:
